### ISSUE TITLE
Single alert for ArgoCdAppOutOfSync alert

### ISSUE THREAD

**Issue by ashish1099** [adinhodovic/argo-cd-mixin#5]
Currently we are getting multiple alerts for each app, it seems little too much, when we have multiple cluster and each cluster has some app outofsync

```
sum by (job, dest_server, project, sync_status) (argocd_app_info{job=~".*",sync_status!="Synced"}) > 0
```

The above expression can make life a little easier, with 1 single alert 
let me know I can raise a PR

**Comment by adinhodovic** [comment]
Is this a HA setup with multiple replicas, where each creates an alert due to `dest_server` being unique?

**Comment by ashish1099** [comment]
@adinhodovic nope, the app name being unique creates the multiple alerts

**Comment by adinhodovic** [comment]
> @adinhodovic nope, the app name being unique creates the multiple alerts

Gotcha, think I'll not add this by default since the other alerts `ArgoCdAppSyncFailed`, `ArgoCdAppAutoSyncDisabled` are app based. Maybe a flag to toggle the behaviour off and on? A flag in config.libsonnet to toggle Alerts by app vs alerts by controller

**Comment by ashish1099** [comment]
yes, I saw that each alert written will endup with multiple alerts, so a toggle make sense or a maybe a record to have name labels in 1 single record and alert on that, so we get the app name (which is what the deal breaker IMHO)

**Comment by adinhodovic** [comment]
> yes, I saw that each alert written will endup with multiple alerts, so a toggle make sense or a maybe a record to have name labels in 1 single record and alert on that, so we get the app name (which is what the deal breaker IMHO)

Yeah, would be open to merging either, with a toggle to not have overlapping alerts.

**Comment by ashish1099** [comment]
I will try this out and send a PR

**Comment by adinhodovic** [comment]
Fixed @ashish1099 !
